tests/integration/utils.py

```diff
@@ -0,0 +1,116 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import hashlib
+import json
+import logging
+import os
+from dataclasses import dataclass
+from pathlib import Path
+from subprocess import PIPE, Popen
+from typing import Any, Dict, List, Tuple
+
+from git import Repo
+
+log = logging.getLogger(__name__)
+
+
+@dataclass
+class TestParameters:
+    repo: str
+    ref: str
+    packages: Tuple[Dict]
+    check_output_json: bool
+    check_deps_checksums: bool
+    expected_rc: int
+    expected_output: str
+
+
+class ContainerImage:
+    def __init__(self, repository: str):
+        """Initialize ContainerImage object with associated repository."""
+        self.repository = repository
+
+    def pull_image(self):
+        cmd = ["podman", "pull", self.repository]
+        output, rc = run_cmd(cmd)
+        if rc != 0:
+            raise RuntimeError(f"Pulling {self.repository} failed. Output:{output}")
+        log.info("Pulled image: %s.", self.repository)
+
+    def run_cmd_on_image(self, cmd: List, tmpdir: Path) -> Tuple[Any, int]:
+        image_cmd = ["podman", "run", "-v", f"{tmpdir}:{tmpdir}:z", self.repository] + cmd
```

Comment:
Could we add `--rm` to the command? Currently, it leaves dead containers lying around after the test finishes

Reply:
sure